In [1]:
import pandas as pd

df = pd.read_csv("usarrests.csv").rename(columns={"Unnamed: 0": "State"})
df.head()

,State,Murder,Assault,UrbanPop
0,Alabama,13.2,236.0,58
1,Alaska,10.0,263.0,48
2,Arizona,8.1,294.0,80
3,Arkansas,8.8,190.0,50
4,California,9.0,276.0,91


In [2]:
summary = df[["Murder", "Assault", "UrbanPop"]].describe().loc[
    ["min", "25%", "mean", "50%", "75%", "max"]
].round(2)
summary

,Murder,Assault,UrbanPop
min,0.80,45.00,6.00
25%,4.08,109.00,53.25
mean,7.79,182.18,74.20
50%,7.25,159.00,66.00
75%,11.25,249.00,77.75
max,17.40,879.00,570.00


In [3]:
flagged = df[(df["UrbanPop"] > 100) | (df["Assault"] > 500) | df.isna().any(axis=1)]
flagged

,State,Murder,Assault,UrbanPop
9,Georgia,17.4,NaN,60
14,Iowa,2.2,56.0,570
39,South Carolina,14.4,879.0,48


**Iowa - UrbanPop = 570**

This figure can’t be right since the percentage for the urban population can’t be more than 100%. Deleting the entire row would’t be my choice since it might be important to check first if there really is an error with the figure.

**South Carolina - Assault = 879**

Even though the figure is very high, it’s not impossible because the assault column is an arrest rate per 100,000 people. I would retain the number provided that the original data source is consulted to confirm if it’s really the number observed or an input error.

**Georgia - Assault is missing**

This figure is not impossible ; however, this is a questionable missing figure that needs some further investigation. I would keep it as missing till I can confirm its original value as opposed to making an assumption about it.

**GitHub repository:** https://github.com/washingtonjustin02smc/CS82A-repository